src/compile/common.ts

```diff
@@ -114,9 +139,6 @@ export function formatSignalRef(fieldDef: FieldDef<string>, specifiedFormat: str
 export function numberFormat(fieldDef: FieldDef<string>, specifiedFormat: string, config: Config, channel: Channel) {
   // Specified format in axis/legend has higher precedence than fieldDef.format
   const format = specifiedFormat;
-  if (fieldDef.type === QUANTITATIVE) {
```

Comment:
I think the type can be O and we might still want to return a format. Look through some specs and see what makes the most sense. Also, we will soon change the default away from `s`. I don't think it should make a difference to you, though.